src/main/java/org/jfrog/hudson/XrayScanResultAction.java

```diff
@@ -0,0 +1,39 @@
+package org.jfrog.hudson;
+
+import hudson.model.BuildBadgeAction;
+import hudson.model.Run;
+import org.apache.commons.lang.StringUtils;
+
+public class XrayScanResultAction implements BuildBadgeAction {
+
+    private final Run build;
+    private String url;
+
+    public XrayScanResultAction(String xrayUrl, Run build) {
+        this.build = build;
+        this.url = xrayUrl;
+    }
+
+    public Run getBuild() {
+        return build;
+    }
+
+    public String getIconFileName() {
+        return "/plugin/artifactory/images/xray-icon.png";
+    }
+
+    public String getDisplayName() {
+        return "Xray Scan  Result";
```

Comment:
Redundant white-space here.
I also suggest changing the display name to *Xray scan report*.